Malmo/src/TCPServer.cpp

```diff
@@ -89,17 +100,38 @@ namespace malmo
             boost::asio::placeholders::error));
     }
     
-    void TCPServer::handleAccept( 
+    void TCPServer::handleAccept(
         boost::shared_ptr<TCPConnection> new_connection,
         const boost::system::error_code& error)
     {
         if (!error)
         {
-            new_connection->read();
-            this->startAccept();
+            if (this->closing)
+            {
+                new_connection.get()->getSocket().close();
+                if (scope != nullptr) 
+                    scope->release();
```

Comment:
Do we also need to set `scope = nullptr` when we call `release()`? Otherwise, is there a danger that `scope` will end up as a dangling pointer? (Or should `this->scope` be a shared_ptr so that happens automatically?)

Reply:
The scope var is a shared pointer and it's possible and very likely that scope is the current object which is being released so should not write to it after the release.

Reply:
Looking closer, this seems to be fine, since `this->startAccept()` doesn't get called if `scope` has been released, so there should be no danger of `scope` being used after that.

Reply:
The scope var _isn't_ a shared pointer - it's a normal pointer to the object that owns us - unless I'm confused?

Reply:
Here it's a shared pointer being released (ref counted) but not in TCPServer (a normal pointer to a SharedScope).

Reply:
This _is_ TCPServer! But anyway, I think it's fine, I can't say any way that `scope` could get reused after `release()` is called. So feel free to ignore this whole thread!